precompiles/vesting/integration_test.go

```diff
@@ -647,157 +767,181 @@ var _ = Describe("Interacting with the vesting extension", func() {
 						WithErrContains(vestingtypes.ErrNotSubjectToClawback.Error())
 				}
 
-				_, _, err = contracts.CallContractAndCheckLogs(s.ctx, s.app, clawbackArgs, clawbackCheck)
-				Expect(err).To(HaveOccurred(), "error while calling the contract: %v", err)
-				if callType.directCall {
-					Expect(err.Error()).To(ContainSubstring("%s: %s", sdk.AccAddress(differentAddr.Bytes()), vestingtypes.ErrNotSubjectToClawback.Error()))
-				}
+				_, _, err = s.factory.CallContractAndCheckLogs(s.keyring.GetPrivKey(0), txArgs, callArgs, clawbackCheck)
+				Expect(err).NotTo(HaveOccurred(), "error while calling the contract: %v", err)
 			})
 
 			It(fmt.Sprintf("should succeed and return empty Coins when all tokens are vested (%s)", callType.name), func() {
 				// commit block with time so that vesting has ended
-				ctx, err := evmosutil.CommitAndCreateNewCtx(s.ctx, s.app, time.Hour*24, nil)
+				err := s.network.NextBlockAfter(time.Hour * 24)
 				Expect(err).ToNot(HaveOccurred(), "error while committing block: %v", err)
-				s.ctx = ctx
 
-				balancePre := s.app.BankKeeper.GetBalance(s.ctx, toAddr.Bytes(), s.bondDenom)
-				Expect(balancePre.Amount).To(Equal(math.NewInt(1100)), "expected different balance after setup")
+				res, err := s.grpcHandler.GetBalance(s.keyring.GetAccAddr(vestingAccIdx), s.bondDenom)
+				Expect(err).To(BeNil())
+				balancePre := res.Balance
 
-				clawbackArgs := s.BuildCallArgs(callType, contractAddr).
-					WithMethodName(vesting.ClawbackMethod).
-					WithArgs(
-						s.address,
-						toAddr,
-						s.address,
-					)
+				callArgs, txArgs := s.BuildCallArgs(callType, contractAddr)
+				callArgs.MethodName = vesting.ClawbackMethod
+				callArgs.Args = []interface{}{
+					s.keyring.GetAddr(0),
+					s.keyring.GetAddr(vestingAccIdx),
+					s.keyring.GetAddr(0),
```

Comment:
could be more verbose IMO by specifying the purpose of the used keys. Might be more code but makes the tests more readable I think by defining e.g. `funder := s.keyring.GetKey(0)` 

```suggestion
					funder.Addr,
					vestingAccount.Addr,
					funder.Addr,
```